libr/util/lib.c

```diff
@@ -73,120 +58,104 @@ R_API void *r_lib_dl_open(const char *libname) {
 		if (!GetModuleFileName (NULL, libname_, MAX_PATH)) {
 			libname_[0] = '\0';
 		}
-	} else {
-		libname_ = r_sys_conv_utf8_to_win (libname);
-	}
-	ret = DLOPEN (libname_);
-#else
-	if (!libname || !*libname) {
-		return NULL;
-	}
-	ret = DLOPEN (libname);
-#endif
-	if (__has_debug && !ret) {
-#if __UNIX__
-		eprintf ("dlerror(%s): %s\n", libname, dlerror ());
-#else
-		eprintf ("r_lib_dl_open: Cannot open '%s'\n", libname);
-#endif
 	}
-#if __WINDOWS__
+	ret = LoadLibrary (libname_);
 	free (libname_);
 #endif
+	if (!ret) {
+		eprintf ("r_lib_dl_open: error: %s (%s)\n", libname, errmsg);
+	}
 	return ret;
 }
 
 R_API void *r_lib_dl_sym(void *handler, const char *name) {
-	return DLSYM (handler, name);
+#if __UNIX__
+	return dlsym (handler, name);
+#elif __WINDOWS__
+	return GetProcAddress (handler, name);
+#else
+	return NULL;
+#endif
 }
 
 R_API int r_lib_dl_close(void *handler) {
-	int ret = -1;
-	if (handler) {
-		ret = DLCLOSE (handler);
-	}
-	return ret;
+#if __UNIX__
+	return dlclose (handler);
+#else
+	return handler? 0: -1;
+#endif
 }
 
-/* ---- */
-
 R_API char *r_lib_path(const char *libname) {
-	char *next, *path0, libpath[1024];
 #if __APPLE__
 	char *env = r_sys_getenv ("DYLD_LIBRARY_PATH");
-	const char *ext = ".dylib";
 	env = r_str_append (env, ":/lib:/usr/lib:/usr/local/lib");
 #elif __UNIX__
 	char *env = r_sys_getenv ("LD_LIBRARY_PATH");
-	const char *ext = ".so";
 	env = r_str_append (env, ":/lib:/usr/lib:/usr/local/lib");
 #else
-	char *env = strdup (".:../../../../../../../windows/system32");
-	const char *ext = ".dll";
+	char *env = strdup (".:../../../../../../../../../../../../windows/system32");
```

Comment:
Wtf?
Use `C;\\Windows\\System32` no?

Reply:
No, the path can be different. It is better to use the API to get the address: https://stackoverflow.com/questions/3544185/how-to-get-the-full-path-to-system32-directory-with-c

Reply:
@GustavoLCR @pelijah can you please add and test a proper way to get the full path here?